medusa/metadata/generic.py

```diff
@@ -173,6 +179,14 @@ def get_poster_path(self, show_obj):
     def get_banner_path(self, show_obj):
         return os.path.join(show_obj.location, self.banner_name)
 
+    def get_image_path(self, show_obj, image_type):
+        """Based on the image_type (banner, poster, fanart) call the correct method, and return the path."""
+        banner_path = {
+            BANNER: self.get_banner_path, POSTER: self.get_poster_path, FANART: self.get_fanart_path
+        }
+        if banner_path.get(image_type):
+            return banner_path.get(image_type)(show_obj)
```

Comment:
You can avoid the second `.get()` here and just use `banner_path[image_type](show_obj)`

Reply:
true